CMakePCHCompiler.cmake

```diff
@@ -233,12 +256,21 @@ function(__watch_pch_last_hook variable access value)
 				string(REPLACE "${flags}" "" value "${value}")
 			endif()
 			if(NOT value STREQUAL "value-NOTFOUND")
-				set_target_properties(
-					"${pch_target}"
-					PROPERTIES
-					"${property}"
-					"${value}"
-					)
+				if(property STREQUAL "CXX_STANDARD")
+					if(NOT MSVC)
+						target_compile_options(
+							"${pch_target}"
+							PUBLIC "-std=gnu++${value}"
```

Comment:
This is a hack. I couldn't find a way to figure out how the actual -std=* string is generated.